syntax = "proto3";

// The purpose of the .proto file is to serve as a contract between the client and the server.
// It defines the data structures and services that are used by the gRPC application.

service Calculator {
  rpc Add (AddRequest) returns (AddResponse);
  rpc Subtract(SubtractRequest) returns (SubtractResponse);
  rpc Multiply(MultiplyRequest) returns (MultiplyResponse);
  rpc Divide(DivideRequest) returns (DivideResponse);
}

// Requests section:
message AddRequest {
  int32 operand1 = 1;
  int32 operand2 = 2;
}

message SubtractRequest {
  int32 operand1 = 1;
  int32 operand2 = 2;
}

message MultiplyRequest {
  float operand1 = 1;
  float operand2 = 2;
}

message DivideRequest {
  float operand1 = 1;
  float operand2 = 2;
}

// Responses section:
message AddResponse {
  int32 result = 1;
}

message SubtractResponse {
  int32 result = 1;
}

message MultiplyResponse {
  float result = 1;
}

message DivideResponse {
  float result = 1;
}
